core/src/main/java/org/jboss/logmanager/LoggerNode.java

```diff
@@ -386,7 +389,22 @@ void publish(final ExtLogRecord record) {
         } catch (VirtualMachineError e) {
             throw e;
         } catch (Throwable t) {
-            // todo - error handler
+            ErrorManager errorManager = handler.getErrorManager();
+            if (errorManager != null) {
+                Exception e;
+                if (t instanceof Exception) {
+                    e = (Exception) t;
+                } else {
+                    e = new UndeclaredThrowableException(t);
+                    e.setStackTrace(EMPTY_STACK);
+                }
+                try {
+                    errorManager.error("Handler publication threw an exception", e, ErrorManager.WRITE_FAILURE);
+                } catch (Throwable t2) {
+                    System.err.println("Handler.reportError caught:");
```

Comment:
Minor change. We should probably use the `StandardOutputStreams.printError()` or `StandardOutputStreams.err.println()`. Same in the `e.printStackTrace(StandardOutputStreams.err)`. The only reason for this is that it captures the `System.out` and `System.err` before they're wrapped with a logger in WildFly.

Reply:
The reason I did this in this way is that it's exactly the same as how `java.util.logging.Handler` reports errors.  I don't think it would be good to do two different things; I think we ought to solve both if possible (but I don't know if it's possible).

Reply:
The issue would be if it hands off to a logger we could end up in an endless loop given the logger may publish to the handler throwing errors.

Reply:
Perhaps we could detect the calling class from the stdio redirector.

Reply:
We do that already so I guess the endless loop isn't really the issue. But currently everywhere we use the captured print stream. I suppose it doesn't matter much, but it just different than what we use elsewhere.

Reply:
Fixed.